core-primitives/time-utils/src/lib.rs

```diff
@@ -60,10 +63,19 @@ mod tests {
 	use super::*;
 
 	#[test]
-	fn subsequent_nows_are_increasing_in_time() {
+	fn subsequent_nows_with_sleep_inbetween_are_strictly_increasing_in_time() {
 		let before = duration_now();
+		thread::sleep(Duration::from_nanos(1));
 		let now = duration_now();
 
 		assert!(before < now);
 	}
+
+	#[test]
+	fn subsequent_nows_are_equal_or_increasing_in_time() {
+		let before = duration_now();
+		let now = duration_now();
+
+		assert!(before <= now);
+	}
```

Comment:
With the information from above, about `SystemTime`, I think this test does not make sense any more 😞 since subsequent `SystemTime::now` calls can be reversed in order

Reply:
Do we even need to have a test in this case? If there's ever an implementation that relies heavily on the ordering, I guess we need to restrict compiler optimizations anyway.

Reply:
Yes, maybe the simplest solution here would be to just remove both tests for now. I don't think we can write a robust test under these conditions. But maybe add a note to our implementation of `now`, that subsequent calls are not guaranteed to be monotonic with a link to https://doc.rust-lang.org/std/time/struct.SystemTime.html

Reply:
Removed them.